src/js/components/FileInput/FileInput.js

```diff
@@ -80,11 +80,31 @@ const FileInput = forwardRef(
     const [hover, setHover] = React.useState();
     const [dragOver, setDragOver] = React.useState();
     const aggregateThreshold = (multiple && multiple.aggregateThreshold) || 10;
+    const max = multiple?.max || Number.POSITIVE_INFINITY;
     const inputRef = useForwardedRef(ref);
     const controlRef = useRef();
     const removeRef = useRef();
     const RemoveIcon = theme.fileInput.icons.remove;
 
+    formContext.useFormField(
+      multiple?.max
+        ? {
+            name,
+            validate: (fileList) => {
```

Comment:
PR #5457 basically added more capability to useFormInput and uses it in FileInput to provide a max file size validation. The `validate` option can be an array of validate functions (first failing function wins.) This PR needs to be adjusted to basically provide both the file count limit and maxSize validation functions in an array to `validate`.

Feel free to reach out if you want my help with that.